include/deal.II/particles/particle_accessor.h

```diff
@@ -282,21 +309,26 @@ namespace Particles
      * internal structure may change this is not intended for public use and
      * only a convenience function for internal purposes.
      */
-    const Particle<dim, spacedim> &
-    get_particle() const;
+    const typename PropertyPool<dim, spacedim>::Handle &
+    get_handle() const;
 
     /**
      * Non-@p const version of the function with the same name above.
      */
-    Particle<dim, spacedim> &
-    get_particle();
+    typename PropertyPool<dim, spacedim>::Handle &
+    get_handle();
 
     /**
      * A pointer to the container that stores the particles. Obviously,
      * this accessor is invalidated if the container changes.
      */
     particle_container *particles;
 
+    /**
+     * A pointer to the property pool that stores the actual particle data.
+     */
+    PropertyPool<dim, spacedim> *property_pool;
```

Comment:
If you wanted to further shrink this structure, store a pointer to the `ParticleHandler` and get at the `particles` and `property_pool` through that pointer.

Reply:
Yes I thought about that, but at the moment the headers are included the other way around (particle handler includes particle accessor) and since all of particle accessor's code is inside the header I can not use a forward declaration. I could add a particle_accessor.cc, but was not sure if it is worth it.